package.json

```diff
@@ -17,35 +17,37 @@
     "pinch-zoom",
     "css3"
   ],
-  "main": "dist/index.js",
+  "main": "./dist/cjs/index.js",
+  "module": "./dist/esm/index.js",
+  "types": "./dist/esm/index.d.ts",
   "scripts": {
-    "build": "babel src/index.jsx -d dist",
+    "build": "yarn build:esm && yarn build:cjs",
+    "build:esm": "tsc",
+    "build:cjs": "tsc --module commonjs --outDir dist/cjs",
+    "build-watch": "yarn build:esm -w && yarn build:cjs -w",
```

Comment:
Even if I also like yarn, I would prefer to stay with npm for simplicity purposes.